lab-taylor/lib/client-pool.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const commands = require('./commands');
+
+function generateNickname() {
+  const nicknameBase = ['monkey', 'slug', 'hippo', 'yeti', 'unicorn'];
+  const idx = Math.floor(Math.random() * ( nicknameBase.length - 0));
```

Comment:
For this assignment, this was fine, I suggest using the node-uuid library for generating ids or unique alphanumeric strings going forward.